### ISSUE TITLE
Investigate process clustering and management options

### ISSUE THREAD

**Issue by chrismytton** [mysociety/popit#770]
Currently if the node process becomes slow or unresponsive then the whole site suffers. We should look into some kind of process clustering so there are several node processes serving requests, which would add some redundancy.

In parallel with clustering we should also look at process management. If one of the processes does become slow or unresponsive it can be removed from the cluster, restarted, and the added back into the cluster once it's healthy again.

**Comment by chrismytton** [comment]
Some potentially useful links:

- Node core's cluster module http://nodejs.org/api/cluster.html
- Carbon Five article about running node in production, contains some good info about zero downtime restarts and more http://blog.carbonfive.com/2014/06/02/node-js-in-production/
- The official joyent node docs on deploying https://www.joyent.com/developers/node/deploy
- The naught module, which wraps up a lot of the clustering stuff into a single package https://github.com/andrewrk/naught

**Comment by chrismytton** [comment]
Closing this in favour of using passenger, see https://github.com/mysociety/popit/issues/659
